sdk/core/Azure.Core/src/Shared/LruCache.cs

```diff
@@ -1,10 +1,12 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 using System.Collections;
 using System.Collections.Generic;
```

Comment:
Would it make sense to use types from `System.Collections.Concurrent` instead of the non-threadsafe variants and our own locking?

Reply:
@JoshLove-msft may have more details because he is the author of this, but I don't think `System.Collections.Concurrent` has any types that work like a list / would work well here, so I'm not sure it would be a direct swap. I also think this approach is the most performant.

Reply:
That's right, the data structure we need is pretty custom and there isn't something that would do what we need from System.Collections.Concurrent.

Reply:
Cool, makes sense.  I was wondering if you could sub out the Dictionary for a ConcurrentDictionary, but I see now you need atomicity across multiple data structures.

I don't have a deep understanding of the custom functionality, but could you achieve what you're trying to do with a ConcurrentQueue?  Probably not, if you need the O(1) lookup time of a Dictionary?  If the functionality is the same minus the lookup time, it might be interesting to compare implementations to understand where the tradeoff is in a linear search vs. dictionary lookup, per Chris's investigation showing that linear searches can actually be pretty fast.  

But this is mostly curiosity from my side.

Reply:
I would avoid attempting to optimize the dictionary look ups as this is not at all a hot path. To be fair, this isn't _that_ custom - it is a simple LRU cache, but there are no existing types in the BCL that support this.